### ISSUE TITLE
createElement doesn't work 

### ISSUE THREAD

**Issue by jmrenouard** [flack/quickform#2]
Hello,

We have a simple example using group:

...
$group[] = HTML_QuickForm::createElement('text', 'first', 'First');
$group[] = HTML_QuickForm::createElement('text', 'last', 'Last');

$form->addGroup($group, 'name', 'Nom:', ',&nbsp;');

But this version doesn't work !
result is:
Fatal error: Using $this when not in object context in C:\xampp\htdocs\ndf\vendor\openpsa\quickform\lib\HTML\QuickForm.php on line 472

Using in a static context $this is never set so we obtain a ugly error message.

The simple fix is changing the context into a static context line 472 in QuickForm.php
$element =& $this->_loadElement('createElement', $elementType, array_slice($args, 1));
become:
$element =& self::_loadElement('createElement', $elementType, array_slice($args, 1));

Then It works perfectly :)

Bye,
Jean-Marie Renouard

**Comment by DerVO** [comment]
I stumbled on the same issue and tried to change `createElement` to `static`. But then I also have to change `_loadElement` to `static` and this is called non-statically from `addElement` and hands an instance of `$this` to `onQuickFormEvent`. Quite a mess to call this statically even in the old PEAR world.

To address this, I dived into these `onQuickFormEvent` functions to figure out what is done there. And it is complex. Depending on the type of event it assumes `$caller` to be an element or the form itself. This would be quite hard to rewrite.

Especially in `HTML_QuickForm_select::onQuickFormEvent()` there is a call of `$caller->isSubmitted()`, so this needs to be an instance of `HTML_QuickForm`. Therefore `_loadElement` can't be static and neither `` `createElement`.

Long story short, this whole Events-thing is a mess and not rewritable easily. So, I also ended up rewriting my calls of `createElement` to non-static ones.

**Comment by flack** [comment]
Yes, it was the same for me. In the end, I think the examples in the old pear doc are just incorrect.
